### DISCUSSION TITLE
Use git tags instead of a `release` branch

### DISCUSSION THREAD

**Discussion by tristan-f-r** [cyberbotics/webots#6377] [Ideas]
Presuming that:
- The `release` branch is to keep track of the latest stable release
- The `master` branch is the development branch

By using git tags, the `sync_protected_branches` action can be dropped.

**Comment by omichel** [comment]
Thank you for sharing this nice idea. However, things are a little bit more complex than what you understood. Basically, we have 3 branches:
- `master` is the branch were we accumulate bug fixes with respect to the current release. This branch is used to possibly create revisions of the current release, e.g., `R2023b-rev1`, `R2023b-rev2`, etc.
- `develop` is the branch were are introducing new features. This branch is used to create the next release, e.g, `R2024a`.
- `released` (and not `release`) is the branch corresponding to the current release of Webots, plus it contains fixes and improvements on the documentation. This branch is actually used to display the online documentation of the current release at https://cyberbotics.com/doc/guide/index

In addition to branches we also have tags. Basically we have a tag for each version of Webots, e.g., `R2023b`, `R2023a`, `R2022b`, `R2022b-rev1`, etc. These tags are bumped on the `master` branch at the time of a release of Webots.

It is convenient to keep the `released` branch instead of a tag to allow us to fix the documentation after a release is published without having to create a new release just for improving the documentation.

Thus, I am not sure how your advice could fit within this picture and how we could drop the branch synchronization actions which basically help us synchronize the `released`, `master` and `develop` branches, e.g., any change made in `released` should be back-ported to `master` and any change in `master` should be back-ported to `develop`.
